README.md

```diff
@@ -393,6 +395,76 @@ BENCHMARK_REGISTER_F(MyFixture, BarTest)->Threads(2);
 /* BarTest is now registered */
 ```
 
+
+## User-defined counters
+
+You can add your own counters with user-defined names. The example below
+will add columns "Foo", "Bar" and "Baz" in its output:
+
+```c++
+static void BM_UserCountersExample(benchmark::State& state) {
+  double numFoos = 0, numBars = 0, numBazs = 0;
+  while (state.KeepRunning()) {
+    // ... count Foo,Bar,Baz events
+  }
+  state.SetCounter("Foo", numFoos);
+  state.SetCounter("Bar", numBars);
+  state.SetCounter("Baz", numBars);
+}
+```
+
+If you are compiling in C++11 mode or later, then overloads with
+std::initializer_list will be enabled. The calls to `SetCounter()` in the
+example above could then be written instead in a single call:
+
+```c++
+  state.SetCounter({{"Foo", numFoos}, {"Bar", numBars}, {"Baz", numBazs}});
+```
+
+`state.SetCounter()` works
+as expected with new or previously existing counters: if the counter already
+exists, then its value will be overwritten. If the counter does not exist,
+then `state.SetCounter()` will create a counter initialized with the provided
+value.
+
+You can mark counters as rates and/or as per-thread averages:
+
+```c++
+  // Set the counter as a rate. It will be presented divided
+  // by the duration of the benchmark.
+  state.SetCounter("FooRate", numFoos, benchmark::Counter::kIsRate);
+
+  // Set the counter as a thread-average quantity. It will
+  // be presented divided by the number of threads. */
+  state.SetCounter("FooRate", numFoos, benchmark::Counter::kAvgThreads);
+
+  // the third argument is a bitwise mask, so you can combine.
+  // This will mark as a thread-average rate:
+  state.SetCounter("FooRate", numFoos, benchmark::Counter::kIsRate|benchmark::Counter::kAvgThreads);
+```
+
+Note that setting the counter flags only works on the first call to `state.SetCounter()`; any subsequent values will be silently ignored.
```

Comment:
We should absolutely diagnose this in debug mode, but probably in release mode as well.